File: java/747. Largest Number At Least Twice of Others/Solution.java
class Solution {
    public int dominantIndex(int[] nums) {

        int index = -1;
        int largest = Integer.MIN_VALUE;
        int secondLargest = Integer.MIN_VALUE;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > largest) {
                secondLargest = largest;
                largest = nums[i];
                index = i;
            } else if (nums[i] > secondLargest){
                secondLargest = nums[i];
            }

        }
        return (largest >= 2 * secondLargest) ? index : -1;
    }
}

File: python/747. Largest Number At Least Twice of Others/solution.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:

        index = -1
        largest = float(-inf)
        second = float(-inf)

        for i in range(len(nums)):
            if nums[i] > largest:
                second = largest
                largest = nums[i]
                index = i
            elif nums[i] > second:
                second = nums[i]


        return index if largest >= 2 * second else -1
    